src/features/SolarSystem/SolarRender.h

```diff
@@ -0,0 +1,345 @@
+#pragma once
+
+#include <Renderer.h>
+#include <gl\GL.h>
+#include <gl\GLU.h>
+#include"features\SolarSystem\SolarResource.h"
+namespace Features
+{
+	using namespace Interfaces;
+
+	class SolarSystemRenderer : public IRenderer
+	{
+	private:
+		float		g_Angle;
+		float 		g_Mercury;
+		float 		g_Venus;
+		float 		g_Earth;
+		float 		g_Mars;
+		float 		g_Jupiter;
+		float 		g_Saturn;
+		float 		g_Uranus;
+		float 		g_Neptune;
+		GLuint		gbTexture[10];
+		GLUquadric	*pQuadric[9];
+	public:
+		void update()
+		{
+			g_Angle = g_Angle + 0.1f;
+			if (g_Angle >= 360)
+				g_Angle = 0.1f;
+
+			g_Mercury = g_Mercury + 0.7f;
+			g_Venus = g_Venus + 0.3f;
+			g_Earth = g_Earth + 0.5f;
+			g_Mars = g_Mars + 0.3f;
+			g_Jupiter = g_Jupiter + 0.5f;
+			g_Saturn = g_Saturn + 0.6f;
+			g_Uranus = g_Uranus + 0.7f;
+			g_Neptune = g_Neptune + 0.5f;
+		}
+		int LoadTexture(GLuint *texture, TCHAR resource[])
+		{
+			HBITMAP hBitmap;
+			BITMAP bmp;
+			int iStatus = FALSE;
+
+			glGenTextures(1, texture);
+			hBitmap = (HBITMAP)LoadImage(GetModuleHandle(NULL), resource, IMAGE_BITMAP, 0, 0, LR_CREATEDIBSECTION);
+			if (hBitmap)
+			{
+				iStatus = TRUE;
+				GetObject(hBitmap, sizeof(bmp), &bmp);
+				glBindTexture(GL_TEXTURE_2D, *texture);
+				glPixelStorei(GL_UNPACK_ALIGNMENT, 4);
+				glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+				glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
+
+				gluBuild2DMipmaps(GL_TEXTURE_2D, 3, bmp.bmWidth, bmp.bmHeight, GL_BGR_EXT, GL_UNSIGNED_BYTE, bmp.bmBits);
+
+				DeleteObject(hBitmap);
+			}
+
+			return(iStatus);
+		}
+
+
+		const char* GetName() override
+		{
+			static const char name[] = "SolarSystemRenderer";
+			return name;
+		}
+
+		RendererResult Initialize(Window window) override
+		{
+			glEnable(GL_TEXTURE_2D);
+			g_Mercury = 0.0f;
+			g_Venus = 0.0f;
+			g_Earth = 0.0f;
+			g_Mars = 0.0f;
+			g_Jupiter = 0.0f;
+			g_Saturn = 0.0f;
+			g_Uranus = 0.0f;
+			g_Neptune = 0.0f;
+			g_Angle = 0.0f;
+
+			LoadTexture(&gbTexture[0], MAKEINTRESOURCE(EARTH_BITMAP));
+			LoadTexture(&gbTexture[1], MAKEINTRESOURCE(MERCURY_BITMAP));
+			LoadTexture(&gbTexture[2], MAKEINTRESOURCE(VENUS_BITMAP));
+			LoadTexture(&gbTexture[3], MAKEINTRESOURCE(MARS_BITMAP));
+			LoadTexture(&gbTexture[4], MAKEINTRESOURCE(JUPITER_BITMAP));
+			LoadTexture(&gbTexture[5], MAKEINTRESOURCE(SATURN_BITMAP));
+			LoadTexture(&gbTexture[6], MAKEINTRESOURCE(URANUS_BITMAP));
+			LoadTexture(&gbTexture[7], MAKEINTRESOURCE(NEPTUNE_BITMAP));
+			LoadTexture(&gbTexture[8], MAKEINTRESOURCE(SUN_BITMAP));
+			LoadTexture(&gbTexture[9], MAKEINTRESOURCE(STARS_BITMAP));
+
+
+			for (int i = 0; i < 9; i++)
+				pQuadric[i] = gluNewQuadric();
```

Comment:
shouldn't we check for the return type of the data returned by gluNewQuadric for the initializations? Fail the initialization if any of them are invalid.

Reply:
It is done